ngraph_bridge/grappler/ngraph_optimizer.cc

```diff
@@ -44,21 +44,23 @@ Status NgraphOptimizer::Init(
     const tensorflow::RewriterConfig_CustomGraphOptimizer* config) {
   const auto params = config->parameter_map();
   if (params.count("ngraph_backend")) {
-    config_backend_name = params.at("ngraph_backend").s();
-    NGRAPH_VLOG(3) << config_backend_name;
-    std::vector<std::string> additional_attributes =
-        BackendManager::GetBackendAdditionalAttributes(config_backend_name);
-    for (size_t i = 0; i < additional_attributes.size(); i++) {
-      if (params.count(additional_attributes[i])) {
-        config_map["_ngraph_" + additional_attributes[i]] =
-            params.at(additional_attributes[i]).s();
-        NGRAPH_VLOG(3) << additional_attributes[i] << " "
-                       << config_map["_ngraph_" + additional_attributes[i]];
+    if (params.count("device_id")) {
+      config_backend_name = params.at("ngraph_backend").s();
+      NGRAPH_VLOG(3) << config_backend_name;
+      for (auto i : params) {
+        if (i.first != "ngraph_backend") {
+          config_map["_ngraph_" + i.first] = i.second.s();
+          NGRAPH_VLOG(3) << "Attribute: " << i.first
+                         << " Value: " << config_map["_ngraph_" + i.first];
+        }
       }
+    } else {
+      return errors::Internal(
+          "NGTF_OPTIMIZER: device_id is a required parameter");
```

Comment:
even `ngraph_backend` is a compulsory attr right?

Lets structure this section like this, so its clear what our contract is. Something like:

```
compulsory_attrs = ["ngraph_backend", "devide_id"]
for (auto itr : compulsory_attrs) {
  if params.count("device_id") == 0: {
      return error_status;
  }
}

compulsory_config_map = {}
optional_extra_config_map = {}
for (auto itr : params) {
if (itr in compulsory_attrs) {
  compulsory_config_map[itr] = params[itr];
} else {
  optional_extra_config_map[itr] = params[itr];
}
}

// Note we split config_map into compulsory_config_map and optional_extra_config_map
// So later on we will not need this line
// config_map.erase("ngraph_backend");
```

with this one can add/remove from the list of compulsory attrs easily later on

Reply:
we'd need to make device_id compulsory:
https://github.com/tensorflow/ngraph-bridge/blob/kanvi/remove_additional_attr_check/ngraph_bridge/ops/ngraph_ops.cc#L24

which means when creating the encapsulate op here, we have to pass the device id.
https://github.com/tensorflow/ngraph-bridge/blob/kanvi/remove_additional_attr_check/ngraph_bridge/ngraph_encapsulate_clusters.cc#L338
Note right now device is populated using `cluster_backend`, lets discuss tomorrow how we are going to pass the other compulsory attr `device-id`.